Nil pointer dereference on malformed Kraftfile

### Describe the bug

```console
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xd36a55]

goroutine 1 [running]:
kraftkit.sh/kconfig.KeyValueMap.Override(0x30?, {0xc000012ba8, 0x1, 0xc0006bcb30?})
        /__w/kraftkit/kraftkit/kconfig/config.go:79 +0x55
kraftkit.sh/unikraft/app.NewProjectFromOptions.func9(0xc0004e1590?)
        /__w/kraftkit/kraftkit/unikraft/app/application_options.go:167 +0x96
kraftkit.sh/unikraft/app.NewApplicationFromOptions({0xc0006bd128, 0xb, 0x18f9315?})
        /__w/kraftkit/kraftkit/unikraft/app/application_options.go:34 +0xb7
kraftkit.sh/unikraft/app.NewProjectFromOptions({0x1b2e308, 0xc000414750}, {0xc000412f70?, 0xc0003620e0?, 0x1635cc0?})
        /__w/kraftkit/kraftkit/unikraft/app/project.go:164 +0xf38
kraftkit.sh/cmd/kraft/run.(*runnerProject).Prepare(0xc000414810, {0x1b2e308, 0xc000414750}, 0xc0000afa20, 0xc0006e1340, {0x0?, 0x0?, 0x7f61d69b6c60?})
        /__w/kraftkit/kraftkit/cmd/kraft/run/runner_project.go:75 +0x1b6
kraftkit.sh/cmd/kraft/run.(*Run).Run(0xc0000afa20, 0x0?, {0x0, 0x0, 0x0})
        /__w/kraftkit/kraftkit/cmd/kraft/run/run.go:278 +0x8d5
kraftkit.sh/cmdfactory.bind.func1(0xc0000e3800?, {0x0, 0x0, 0x0})
        /__w/kraftkit/kraftkit/cmdfactory/builder.go:689 +0x19d
kraftkit.sh/cmdfactory.bind.func1(0xc0000e3800?, {0x0, 0x0, 0x0})
        /__w/kraftkit/kraftkit/cmdfactory/builder.go:689 +0x19d
kraftkit.sh/cmdfactory.bind.func1(0xc0000e3800?, {0x0, 0x0, 0x0})
        /__w/kraftkit/kraftkit/cmdfactory/builder.go:689 +0x19d
kraftkit.sh/cmdfactory.bind.func1(0x0?, {0x0, 0x0, 0x0})
        /__w/kraftkit/kraftkit/cmdfactory/builder.go:689 +0x19d
kraftkit.sh/cmdfactory.bind.func1(0xc0006bdd80?, {0x0, 0x0, 0x0})
        /__w/kraftkit/kraftkit/cmdfactory/builder.go:689 +0x19d
kraftkit.sh/cmdfactory.execute(0xc0000e3800, {0x2770e20, 0x0, 0x0})
        /__w/kraftkit/kraftkit/cmdfactory/builder.go:231 +0x30b
kraftkit.sh/cmdfactory.executeC(0xc000004300)
        /__w/kraftkit/kraftkit/cmdfactory/builder.go:291 +0x205
kraftkit.sh/cmdfactory.Main({0x1b2e308?, 0xc0001d0780}, 0xc000004300)
        /__w/kraftkit/kraftkit/cmdfactory/builder.go:322 +0x73
main.main()
        /__w/kraftkit/kraftkit/cmd/kraft/kraft.go:153 +0x549
```

### Steps to reproduce

This `Kraftfile` (missing `=y` after the kconfig key):

```yaml
specification: v0.5

unikraft:
  version: stable
  kconfig:
    - CONFIG_LIBUKDEBUG_PRINTK_INFO

targets:
  - name: default
    architecture: x86_64
    platform: kvm
```

### Expected behavior

Some helpful error.

### Which architectures were you using or does this bug affect?

_No response_

### Which operating system were you using or does this bug affect?

_No response_

### Relevant log output

_No response_

--- a/kconfig/config.go
+++ b/kconfig/config.go
@@ -20,7 +20,7 @@ const DotConfigFileName = ".config"
 type KeyValueMap map[string]*KeyValue
 
 // NewKeyValueMapFromSlice build a new Mapping from a set of KEY=VALUE strings
-func NewKeyValueMapFromSlice(values ...interface{}) KeyValueMap {
+func NewKeyValueMapFromSlice(values ...interface{}) (KeyValueMap, error) {
 	mapping := KeyValueMap{}
 
 	for _, value := range values {
@@ -32,26 +32,29 @@ func NewKeyValueMapFromSlice(values ...interface{}) KeyValueMap {
 			str = fmt.Sprintf("%d", t)
 		}
 		tokens := strings.SplitN(str, "=", 2)
-		if len(tokens) > 1 {
+		if len(tokens) > 1 && tokens[1] != "" {
 			mapping[tokens[0]] = &KeyValue{
 				Key:   tokens[0],
 				Value: tokens[1],
 			}
 		} else {
-			mapping[str] = nil
+			return nil, fmt.Errorf("kconfig option must be a key-value pair(key=value), found: %v", str)
 		}
 	}
 
-	return mapping
+	return mapping, nil
 }
 
 // NewKeyValueMapFromMap build a new Mapping from a set of KEY=VALUE strings
-func NewKeyValueMapFromMap(values map[string]interface{}) KeyValueMap {
+func NewKeyValueMapFromMap(values map[string]interface{}) (KeyValueMap, error) {
 	mapping := KeyValueMap{}
-
 	for key, value := range values {
 		mapping[key] = &KeyValue{
 			Key: key,
+		}
+
+		if value == nil {
+			return nil, fmt.Errorf("kconfig option must have a value, on key: %v", key)
 		}
 
 		switch casting := value.(type) {
@@ -70,7 +73,7 @@ func NewKeyValueMapFromMap(values map[string]interface{}) KeyValueMap {
 		}
 	}
 
-	return mapping
+	return mapping, nil
 }
 
 // Override accepts a list of key value pairs and overrides the key in the map

--- a/unikraft/component/component.go
+++ b/unikraft/component/component.go
@@ -50,6 +50,7 @@ func NameAndVersion(component Component) string {
 // primarily used by other components internally which follow a similar format
 // of specifying a version, source and list of KConfig options.
 func TranslateFromSchema(props interface{}) (map[string]interface{}, error) {
+	var err error
 	component := make(map[string]interface{})
 
 	switch entry := props.(type) {
@@ -79,9 +80,12 @@ func TranslateFromSchema(props interface{}) (map[string]interface{}, error) {
 			case "kconfig":
 				switch tprop := prop.(type) {
 				case map[string]interface{}:
-					component["kconfig"] = kconfig.NewKeyValueMapFromMap(tprop)
+					component["kconfig"], err = kconfig.NewKeyValueMapFromMap(tprop)
 				case []interface{}:
-					component["kconfig"] = kconfig.NewKeyValueMapFromSlice(tprop...)
+					component["kconfig"], err = kconfig.NewKeyValueMapFromSlice(tprop...)
+				}
+				if err != nil {
+					return nil, err
 				}
 			}
 		}

--- a/unikraft/target/transform.go
+++ b/unikraft/target/transform.go
@@ -16,6 +16,7 @@ import (
 )
 
 func TransformFromSchema(ctx context.Context, data interface{}) (interface{}, error) {
+	var err error
 	switch value := data.(type) {
 	case map[string]interface{}:
 		uk := unikraft.FromContext(ctx)
@@ -52,9 +53,12 @@ func TransformFromSchema(ctx context.Context, data interface{}) (interface{}, er
 			case "kconfig":
 				switch tprop := prop.(type) {
 				case map[string]interface{}:
-					t.kconfig = kconfig.NewKeyValueMapFromMap(tprop)
+					t.kconfig, err = kconfig.NewKeyValueMapFromMap(tprop)
 				case []interface{}:
-					t.kconfig = kconfig.NewKeyValueMapFromSlice(tprop...)
+					t.kconfig, err = kconfig.NewKeyValueMapFromSlice(tprop...)
+				}
+				if err != nil {
+					return nil, err
 				}
 			}
 		}

